app/services/oneoffs/npq/backfill_trn_to_teacher_profile.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Oneoffs::NPQ
+  class BackfillTrnToTeacherProfile
+    def migrate
+      # locate all TeacherProfiles with empty TRN
+      teacher_profiles_without_trns.in_batches.each_record do |profile|
+        trns = profile.user.npq_applications.map(&:teacher_reference_number).uniq
+        # check if they are correct
+        if trns.count == 1 && TeacherReferenceNumber.new(trns.first).valid?
+          profile.trn = trns.first
+          profile.save!
+        end
```

Comment:
Can we use the `TeacherReferenceNumber.formatted_trn` here to ensure we store a correctly formatted one? 
```suggestion
        trn = TeacherReferenceNumber.new(trns.first)
        if trns.count == 1 && trn.valid?
          profile.trn = trn.formatted_trn
          profile.save!
        end
```